src/components/SignInForm/SignInForm.jsx

```diff
@@ -0,0 +1,170 @@
+import { useContext, useState, useCallback } from "react";
+import styles from "./SignInForm.module.scss";
+import { AuthContext } from "../../containers/AuthContext";
+
+const SignInForm = () => {
+  const authContext = useContext(AuthContext);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
```

Comment:
we probably don't need this state. 
same can be achieved with `errors`.
kinda: `if errors then do not submit`

Reply:
this state does not allow the user to press the button again while the data is being sent to the back, if it is unnecessary I will delete it

Reply:
nah, we don't need it :)
please check my other comments below.

and have a look into submit handler - this guy checks for errors, 
if we have any errors we return from the handler function, not making api call. 

and there we don't want to disable submit button as well, this is antipattern :)

Reply:
Fixed! Thank you for your thorough explanation!